stock_picking_product_generate_sequence/models/stock_move_line.py

```diff
@@ -0,0 +1,49 @@
+##############################################################################
+# Copyright (c) 2022 brain-tec AG (https://braintec.com)
+# All Right Reserved
+#
+# See LICENSE file for full licensing details.
+##############################################################################
+from re import findall as regex_findall
+from re import split as regex_split
+
+from odoo import models, fields, api
+
+
+class StockMoveLine(models.Model):
+    _inherit = 'stock.move.line'
+
+    def action_generate_product_serial(self):
+        self.ensure_one()
+        self.lot_id = self.env['stock.production.lot'].create({
+            'product_id': self.product_id.id,
+            'company_id': self.company_id.id,
+            'name': self._get_next_serial(self.company_id, self.product_id) or
```

Comment:
[Question] The serial introduced by the user and the one automatically created by the sequence stock.picking.product have different formats? I don't see restrictions for the user to also add it as SPP00000001 Otherwise if they would be the same you should advance the sequence not to get duplicates according to the existing serials

Reply:
Very good point. I add a restriction to check if the sequence product is the same that the global sequence. thx